src/core/font_renderer.js

```diff
@@ -656,7 +669,8 @@ var FontRendererFactory = (function FontRendererFactoryClosure() {
     this.glyphs = glyphs;
     this.cmap = cmap;
 
-    this.compiledGlyphs = [];
+    this.compiledGlyphs = Object.create(null);
+    this.compiledCharCodeToGlyphId = Object.create(null);
```

Comment:
Looking through this once more, I'm assuming that these two lines could actually be removed given the `CompiledFont.call(this, fontMatrix);` line above?